crypto/src/mls/conversation/merge.rs

```diff
@@ -273,107 +218,30 @@ mod tests {
+                        alice_central.context.new_update_proposal(&id).await.unwrap();
 
+                        assert_eq!(alice_central.pending_proposals(&id).await.len(), 3);
+                        // rejecting the next proposal will clear all proposals
                         alice_central
-                            .context
-                            .clear_pending_proposal(&id, update_ref)
-                            .await
-                            .unwrap();
+                            .central
+                            .provide_transport(Arc::new(CoreCryptoTransportAbortProvider::default()))
+                            .await;
+                        alice_central.context.new_update_proposal(&id).await.unwrap_err();
                         assert!(alice_central.pending_proposals(&id).await.is_empty());
-                        assert!(!alice_central
-                            .pending_proposals(&id)
-                            .await
-                            .into_iter()
-                            .any(|p| matches!(p.proposal(), Proposal::Update(_))));
                     })
                 },
             )
             .await
         }
 
-        #[apply(all_cred_cipher)]
-        #[wasm_bindgen_test]
-        pub async fn should_fail_when_conversation_not_found(case: TestCase) {
-            use crate::{mls, LeafError};
-
-            run_test_with_client_ids(case.clone(), ["alice"], move |[alice_central]| {
-                Box::pin(async move {
-                    let id = conversation_id();
-                    let simple_ref = MlsProposalRef::from(vec![0; case.ciphersuite().hash_length()]);
-                    let clear = alice_central.context.clear_pending_proposal(&id, simple_ref).await;
-                    assert!(matches!(clear.unwrap_err(), mls::conversation::error::Error::Leaf(LeafError::ConversationNotFound(conv_id)) if conv_id == id))
-                })
-            })
-            .await
-        }
-
-        #[apply(all_cred_cipher)]
-        #[wasm_bindgen_test]
-        pub async fn should_fail_when_proposal_ref_not_found(case: TestCase) {
-            run_test_with_client_ids(case.clone(), ["alice"], move |[mut alice_central]| {
-                Box::pin(async move {
-                    let id = conversation_id();
-                    alice_central
-                        .context
-                        .new_conversation(&id, case.credential_type, case.cfg.clone())
-                        .await
-                        .unwrap();
-                    assert!(alice_central.pending_proposals(&id).await.is_empty());
-                    let any_ref = MlsProposalRef::from(vec![0; case.ciphersuite().hash_length()]);
-                    let clear = alice_central.context.clear_pending_proposal(&id, any_ref.clone()).await;
-                    assert!(
-                        matches!(clear.unwrap_err(), Error::PendingProposalNotFound(prop_ref) if prop_ref == any_ref)
```

Comment:
I'm not sure if I understand the reason behind removing the `PendingProposalNotFound` enum variant and this test. Could you please provide more details on that?

I'm trying to imagine if there is a case where we get a proposal ref from another client and then try to find it locally.